pkg/command/context.go

```diff
@@ -649,31 +649,98 @@ func globalTanzuLogin(c *configtypes.Context, generateContextNameFunc func(orgNa
 	return errors.New(invalidIdpType)
 }
 
-func globalTanzuLoginUAA(c *configtypes.Context, generateContextNameFunc func(orgName, endpoint string, isStaging bool) string) error {
-	uaaEndpoint := c.AdditionalMetadata[config.TanzuAuthEndpointKey].(string)
-	log.V(7).Infof("Login to UAA endpoint: %s", uaaEndpoint)
+type DeploymentMetadata struct {
+	APIEndpoint  string                         `json:"apiEndpoint"`
+	Organization DeploymentMetadataOrganization `json:"organization"`
+}
 
-	claims, err := doInteractiveLoginAndUpdateContext(c, uaaEndpoint)
+type DeploymentMetadataOrganization struct {
+	DisplayName string `json:"displayName"`
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+}
+
+func getOrgFromSelfManagedEndpoint(c *configtypes.Context) (string, string, error) {
+	hubEndpoint := c.AdditionalMetadata[config.TanzuHubEndpointKey].(string)
```

Comment:
Any possibility that `c.AdditionalMetadata[config.TanzuHubEndpointKey]` is not configured and this causes a panic?

Reply:
good point. To be safe, updated to check for existence as well.